crates/ruma-common/src/events/room/message.rs

```diff
@@ -553,6 +624,27 @@ impl MessageType {
         }
     }
 
+    /// Set the [mentions] of this message.
+    ///
+    /// [mentions]: https://spec.matrix.org/latest/client-server-api/#user-and-room-mentions
+    pub fn set_mentions(&mut self, mentions: Option<Mentions>) {
+        let s_mentions = match self {
+            MessageType::Audio(m) => &mut m.mentions,
+            MessageType::Emote(m) => &mut m.mentions,
+            MessageType::File(m) => &mut m.mentions,
+            MessageType::Image(m) => &mut m.mentions,
```

Comment:
Why add `mentions` to all the `MessageType`s, instead of to `RoomMessageEventContent`?

Reply:
Iirc, it is for `m.replace` relations (edits). We generate an `*EventContentWithoutRelation` for this kind of relation, because it's not possible to change the relations of an event, but with `RoomMessageEventContent` that type is actually `MessageType`. And since it is possible to change the mentions of an event, we need the mentions inside it (or to generate a type that is a `struct { msgtype: MessageType, mentions: Mentions }`).

Reply:
> (or to generate a type that is a `struct { msgtype: MessageType, mentions: Mentions }`).

I would prefer this. It's a little unfortunate that it's a breaking change, but shouldn't be a big problem.

Reply:
Done :slightly_smiling_face: